app/code/Magento/InventorySalesApi/Api/RegisterSalesEventInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySalesApi\Api;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\Data\SalesEventInterface;
+
+/**
+ * This service is responsible for creating reservations upon a sale event.
+ *
+ * @api
+ */
+interface RegisterSalesEventInterface
+{
+    /**
+     * @param string[] $items
+     * @param string[] $productTypes
+     * @param SalesChannelInterface $salesChannel
+     * @param SalesEventInterface $salesEvent
+     * @return void
+     * @throws LocalizedException
+     */
+    public function execute(array $items, SalesChannelInterface $salesChannel, SalesEventInterface $salesEvent);
```

Comment:
Just speculating: "item to sale" expresses the point of view of the merchant but maybe the one who is calling the API is a customer from whose point of view the item is "to buy"... maybe "item to reserve" can work for both because whether you are selling or buying something, you are going to reserve it and indeed after submitting an order a reservation is created; what do you think @maghamed ?

Reply:
approved naming: `ItemToSellInterface`